client/ayon_sitesync/addon.py

```diff
@@ -1140,6 +1141,26 @@ def _transform_sites_from_settings(self, settings):
                 sites[site_name] = configured_site
         return sites
 
+    def _get_project_roots_for_site(self, project_name, local_site_id):
+        """Returns projects roots and their overrides."""
+        # overrides for Studio site for particular user
+        roots = get_project_roots_for_site(project_name, local_site_id)
```

Comment:
Won't this return only overrides? So if user does not have any overriden it will be empty dictionary...

Reply:
This is why there is re-apply of `default_roots` little bit lower. (These contain only default values, no overrides.)

Reply:
But it will not get there if you get empty dictionary or one root with filled values because of `if all(roots.values()):`